mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/ResourceClusterAwareSchedulerActor.java

```diff
@@ -216,6 +220,15 @@ private void onFailedToSubmitScheduleRequestEvent(FailedToSubmitScheduleRequestE
             event.getScheduleRequestEvent().getRequest().getWorkerId(),
             event.getScheduleRequestEvent().getRequest().getStageNum(),
             Throwables.getStackTraceAsString(event.throwable)));
+
+        try {
+            resourceCluster.reconnectTaskExecutorGateway(event.getTaskExecutorID()).join();
```

Comment:
Why make this a blocking call especially if you are not planning to do anything with the result?

Reply:
Also, isn't a reconnection already done as part of getting the task executor RPC connection object?

Reply:
I will merge for now to try on prod. Feel free to leave more comments and I can follow-up.